spyder/api/toolbar.py

```diff
@@ -0,0 +1,146 @@
+        colspan = 1 if row == 0 else 3
+        self.layout.addLayout(row_layout, row, self.CONTENT_COL, 1, colspan)
+
+    def _get_hboxlayout_at_row(self, row):
+        """
+        Return the QHBoxLayout at row row. Add one if it doesn't already
+        exist.
+        """
+        if self.layout.itemAtPosition(row, self.CONTENT_COL) is None:
+            self._add_hboxlayout_at_row(row)
+        return self.layout.itemAtPosition(row, self.CONTENT_COL)
+
+    def set_row_visible(self, row, state):
+        """
+        Set the visibility of all widgets at row row to state.
+        """
+        hboxlayout = self.layout.itemAtPosition(row, self.CONTENT_COL)
+        if (state and not self.isVisible()) or hboxlayout is None:
+            return
+        for index in range(hboxlayout.count()):
+            try:
+                hboxlayout.itemAt(index).widget().setVisible(state)
+            except AttributeError:
+                pass
+
+    def add_item(self, item, stretch=None, row=0):
+        """
+        Add a widget or a layout with an horizontal stretch factor stretch
+        to the end of row row of this toolbar.
+        """
+        row_layout = self._get_hboxlayout_at_row(row)
+
+        if item is None:
+            item = QFrame()
+            item.setFrameShape(QFrame.VLine)
+            item.setFrameShadow(QFrame.Sunken)
+        elif isinstance(item, int):
+            self.add_spacing(spacing=item, row=row)
+
+        column = row_layout.count()
+        if row_layout.itemAt(column - 1) is None:
+            column = max(0, column - 1)
+
+        try:
+            row_layout.insertWidget(column, item)
+        except TypeError:
+            row_layout.insertLayout(column, item)
+
+        if stretch is not None:
+            row_layout.setStretchFactor(item, stretch)
+
+    def add_widget(self, widget, stretch=None, row=0):
+        """
+        Add a widget with an horizontal stretch factor stretch to the end
+        of row row of this toolbar.
+        """
+        self.add_item(widget, stretch=stretch, row=row)
+
+    def add_stretch(self, stretch, row=0):
+        """
+        Add a stretchable space with zero minimum size and stretch factor
+        stretch to the end of this toolbar.
+        """
+        row_layout = self._get_hboxlayout_at_row(row)
+        row_layout.addStretch(stretch)
+
+    def add_spacing(self, spacing=None, row=0):
+        """
+        Add a non-stretchable space with size spacing to the end of
+        this toolbar.
+        """
+        row_layout = self._get_hboxlayout_at_row(row)
+        if spacing is None:
+            spacing = QApplication.instance().style().pixelMetric(
+                QStyle.PM_LayoutHorizontalSpacing)
+        elif spacing == 'label':
+            spacing = QApplication.instance().style().pixelMetric(
+                QStyle.PM_CheckBoxLabelSpacing)
+        row_layout.addSpacing(spacing)
+
+    def add_options_btn(self, options_btn, spacing=None, stretch=1):
```

Comment:
I am personally not a fan of contractions on names :-p 

I suggest we use `add_options_button`